yjit/src/codegen.rs

```diff
@@ -1963,6 +1963,11 @@ fn gen_get_ivar(
     recv_opnd: YARVOpnd,
     side_exit: CodePtr,
 ) -> CodegenStatus {
+    // If the object has a too complex shape, we exit
+    if comptime_receiver.is_too_complex() {
+        return CantCompile;
+    }
```

Comment:
Here I wish we could call a C function for setivar / getivar instead of exiting, but I understand that fixing the outstanding problem is more important.

Reply:
Yes, I agree.  I think we can just fall back to a C function, but I wanted to get the memory usage under control first and I didn't want to make this diff larger.